packages/js/src/utils/generators.ts

```diff
@@ -0,0 +1,39 @@
+/** @link [Proof Key for Code Exchange by OAuth Public Clients](https://datatracker.ietf.org/doc/html/rfc7636) */
+
+import { UrlSafeBase64 } from '@silverhand/essentials';
+import { fromUint8Array } from 'js-base64';
+
+const DEFAULT_RANDOM_DATA_LENGTH = 64;
+
+/**
+ * @param length The length of the raw random data.
+ */
+function generateRandomString(length: number) {
+  const randomValues = crypto.getRandomValues(new Uint8Array(length));
+  const base64String = fromUint8Array(randomValues);
```

Comment:
If use `crypto.getRandomValues(new Uint8Array(length))`, which returns `Uint8Array` data, to generate random data, how to implement base64-encode and base64-decode under both Node and browser-like environments?

( js/js should be able to run under both Node and browser-like environments)

- from `Uint8Array` to base64 string
- from base64 string to `Uint8Array`

Any idea? Ask for help @logto-io/eng

Reply:
![image](https://user-images.githubusercontent.com/10594507/150052389-8e3689ce-6f7f-446a-92c6-847b34cf4f0e.png)